compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -592,14 +592,34 @@ impl<'hir> LoweringContext<'_, 'hir> {
     ) -> hir::ExprKind<'hir> {
         let output = ret_ty.unwrap_or_else(|| hir::FnRetTy::DefaultReturn(self.lower_span(span)));
 
-        // Resume argument type: `ResumeTy`
-        let unstable_span =
-            self.mark_span_with_reason(DesugaringKind::Async, span, self.allow_gen_future.clone());
-        let resume_ty = hir::QPath::LangItem(hir::LangItem::ResumeTy, unstable_span, None);
+        // Resume argument type, which should be `&mut Context<'_>`.
+        // NOTE: Using the `'static` lifetime here is technically cheating.
+        // The `Future::poll` argument really is `&'a mut Context<'b>`, but inferring that
+        // lifetime would lead to lifetime errors, as those lifetimes are considered captured.
```

Comment:
since they are on the generator argument, why would they get captured?

ah this is https://github.com/rust-lang/rust/issues/68923

Please reference that issue here, as we'll lose the connection now that `ResumeTy` is dead code

Reply:
👍🏻 I slightly reworded that comment, referencing the given issue, and have also rebased the PR.